test/modules/http/regressions/http-socket-enetunreach.test.ts

```diff
@@ -0,0 +1,23 @@
+import { it, expect, beforeAll, afterAll } from 'vitest'
+import { httpGet } from '../../../helpers'
+import { ClientRequestInterceptor } from '../../../../src/interceptors/ClientRequest'
+
+const interceptor = new ClientRequestInterceptor()
+
+beforeAll(() => {
+  interceptor.apply()
+})
+
+afterAll(() => {
+  interceptor.dispose()
+})
+
+it('does not leave the test process hanging due to the custom socket timeout', async () => {
+  interceptor.apply();
+  interceptor.on('request', ({ request }) => {
+    request.respondWith(new Response('test'))
```

Comment:
May also be a good idea to create a deferred promise for the `request.url` value here in the request listener to make sure we are providing the correct IPv6 URL in the listener. What do you think?

```ts
const listenerUrlPromise = new DeferredPromise<string)

interceptor.on('request', ({ request }) => {
  listenerUrlPromise.resolve(request.url)
  request.respondWith(new Response('test'))
})

// ...
const listenerRequestUrl = await listenerUrlPromise
expect(listenerRequestUrl).toBe('IPv6_URL_HERE')
```

Would have to move the `interceptor.on()` listener into the test to make this work.

Reply:
Absolutely agree!
Fixed.